libs/ml/include/ml/optimisation/learning_rate_params.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/base_types.hpp"
+#include "math/statistics/mean.hpp"
+#include "ml/dataloaders/dataloader.hpp"
+#include "core/serializers/main_serializer.hpp"
+
+#include <chrono>
+
+namespace fetch {
+namespace ml {
+namespace optimisers {
+/**
+ * Training annealing config
+ * @tparam T
+ */
+template <typename T>
+struct LearningRateParam
+{
+  using DataType = T;
+  enum class LearningRateDecay : uint8_t
+  {
+    EXPONENTIAL,
+    LINEAR,
+    NONE
+  };
+  LearningRateDecay mode                   = LearningRateDecay::NONE;
+  DataType          starting_learning_rate = static_cast<DataType>(0.1);
+  DataType          ending_learning_rate   = static_cast<DataType>(0.0001);
+  DataType          linear_decay_rate      = static_cast<DataType>(0.0000000000001);
+  DataType          exponential_decay_rate = static_cast<DataType>(0.999);
+};
+}  // namespace optimisers
+}  // namespace ml
+
+namespace serializers {
+/**
+ * serializer for conv1d saveable params
```

Comment:
Why is serializer (serialise? for British English?)  for conv1D in file learning_rate_params? Shouldn't it be serialise for optimiser params?

Reply:
I would agree, but unfortunately it is not my choice since it is already serialize everywhere in the repo and it needs to be that namespace. Perhaps someone can put in a PR to replace it everywhere.